File: android/app/src/main/java/ai/mapier/swipe/analytics/PostHogAnalytics.kt
package ai.mapier.swipe.analytics

import android.content.Context
import com.posthog.PersonProfiles
import com.posthog.PostHog
import com.posthog.android.PostHogAndroid
import com.posthog.android.PostHogAndroidConfig

object PostHogAnalytics {
  fun make(
    context: Context,
    settings: AnalyticsSettings,
  ): AnalyticsCapturing {
    if (!settings.isEnabled) return NoOpAnalytics

    val configuration = PostHogAndroidConfig(
      apiKey = settings.projectToken,
      host = settings.host,
    ).apply {
      captureApplicationLifecycleEvents = false
      captureScreenViews = false
      captureDeepLinks = false
      sessionReplay = false
      errorTrackingConfig.autoCapture = false
      sendFeatureFlagEvent = false
      preloadFeatureFlags = false
      setDefaultPersonProperties = false
      personProfiles = PersonProfiles.NEVER
    }
    PostHogAndroid.setup(context, configuration)

    return AnalyticsClient { name, properties ->
      PostHog.capture(
        event = name,
        properties = properties.mapValues { it.value as Any },
      )
    }
  }
}

File: ios/Wavepoint/Analytics/PostHogAnalytics.swift
import Foundation
import PostHog

enum PostHogAnalytics {
  static func make(from bundle: Bundle = .main) -> any AnalyticsCapturing {
    let settings = AnalyticsSettings(
      projectToken: bundle.object(forInfoDictionaryKey: "POSTHOG_PROJECT_TOKEN") as? String ?? "",
      host: bundle.object(forInfoDictionaryKey: "POSTHOG_HOST") as? String ?? ""
    )
    guard settings.isEnabled else { return NoOpAnalytics() }

    let configuration = PostHogConfig(
      projectToken: settings.projectToken,
      host: settings.host
    )
    configuration.captureApplicationLifecycleEvents = false
    configuration.captureScreenViews = false
    configuration.captureElementInteractions = false
    configuration.enableSwizzling = false
    configuration.sendFeatureFlagEvent = false
    configuration.preloadFeatureFlags = false
    configuration.setDefaultPersonProperties = false
    configuration.sessionReplay = false
    configuration.surveys = false
    PostHogSDK.shared.setup(configuration)

    return AnalyticsClient { name, properties in
      PostHogSDK.shared.capture(
        name,
        properties: properties.mapValues { $0 as Any }
      )
    }
  }
}
